draft-barnes-acme.md

```diff
@@ -1140,45 +1140,96 @@ If the server is able to validate the signature and confirm that the jwk and alg
 
 ## DNS
 
-When the identifier being validated is a domain name, the client can prove control of that domain by provisioning records under it.   The DNS challenge requires the client to provision a TXT record containing a validation token under a specific validation domain name.
+When the identifier being validated is a domain name, the client can prove control of that domain by provisioning resource records under it.   The DNS challenge requires the client to provision a TXT record containing a validation token under a specific validation domain name.
 
 type (required, string):
 : The string "dns"
 
 token (required, string):
-: An ASCII string that is to be provisioned in the TXT record.  This string SHOULD be randomly generated, with at least 128 bits of entropy (e.g., a hex-encoded random octet string).
+: An ASCII string that uniquely identifies this challenge
+
+serverPublicKey (required, JWK):
+: An ECDH public key that will be used to generate a shared secret
 
 ~~~~~~~~~~
 
 {
   "type": "dns",
-  "token": "17817c66b60ce2e4012dfad92657527a"
+  "token": "17817c66b60ce2e4012dfad92657527a",
+  "serverPublicKey": { "kty": "EC", ... }
```

Comment:
Agreed, but I think this needs to be done within a broader PR to add examples for the whole thing.  And we probably want to do that only when Boulder supports this, so that we can auto-generate them.